### ISSUE TITLE
Multiple connection strings for master-slave demotion/promotion.

### ISSUE THREAD

**Issue by robblovell** [NodeRedis/node_redis#830]
On ElasticCache, a redis cluster has one primary master read/write node and several secondary slave read-only nodes.  The redis cluster periodically switches roles for nodes, exchanging master and slave nodes. The connection string to the ElasticCache is to the primary node when writes are needed. The createClient should be modified to take more than one connection port and host pair, and try each one in turn when an error like "Can't save packet:Error: READONLY You can't write against a read only slave." occurs so that the client can recover if the primary is switched while in operation.

**Comment by epoberezkin** [comment]
Can be related to #821 - essentially multiple connection settings to support any kind of switching in redis

**Comment by Mattsi-Jansky** [comment]
We have also been bitten by this. Compose's Redis distribution is doing some maintenance soon, and will be failing back to their slave Redis instance during the outage. They recommend using a `host:port,host:port` connection string to maintain availability, but it isn't supported.

**Comment by samuelt1** [comment]
if I made a fork for this would you someone even consider approving it?

**Comment by stockholmux** [comment]
humm.. @Mattsi-Jansky and @samuelt1 could you use the solution outlined #1050 ?

If I'm understanding the solution of just specifying multiple connection strings, I can see cases where not everyone would want this.

**Comment by samuelt1** [comment]
yeah, so that is how I am fixing this on my end. I just have my own function that will connect to all of the servers, and then drop the slaves but not the master.
